numpy/_core/src/multiarray/multiarraymodule.c

```diff
@@ -4572,7 +4572,7 @@ static struct PyMethodDef array_module_methods[] = {
     {"from_dlpack", (PyCFunction)from_dlpack,
         METH_FASTCALL | METH_KEYWORDS, NULL},
     {"_unique_hash",  (PyCFunction)array__unique_hash,
-        METH_O, "Collect unique values via a hash map."},
+        METH_VARARGS | METH_KEYWORDS, "Collect unique values via a hash map."},
```

Comment:
Please use `METH_FASTCALL | METH_KEYWORDS`, you'll have to change the parsing, but you should find an example easily (and the call overhead will be much smaller).